app/src/main/java/com/rahobbs/todo/database/TodoCursorWrapper.java

```diff
@@ -0,0 +1,41 @@
+package com.rahobbs.todo.database;
+
+import android.database.Cursor;
+import android.database.CursorWrapper;
+
+import com.rahobbs.todo.helpers.TodoItem;
+import com.rahobbs.todo.database.TodoSchema.TodoTable;
+
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * Created by rachael on 2/9/16.
+ * Delegates calls to Cursor object
+ */
+public class TodoCursorWrapper extends CursorWrapper {
+    public TodoCursorWrapper(Cursor cursor){
+        super(cursor);
+    }
+
+    public TodoItem getTodoItem(){
```

Comment:
Something like this...

```java
  public TodoItem(Cursor cursor) {
    this(UUID.fromString(cursor.getString(cursor.getColumnIndex(TodoTable.Cols.UUID))));
    mTitle = cursor.getString(cursor.getColumnIndex(TodoTable.Cols.TITLE));
    mDate = new Date(cursor.getLong(cursor.getColumnIndex(TodoTable.Cols.DATE)));
    completed = cursor.getInt(cursor.getColumnIndex(TodoTable.Cols.COMPLETED)) != 0;
    mDetails = cursor.getString(cursor.getColumnIndex(TodoTable.Cols.DETAILS));
    mParents = cursor.getString(cursor.getColumnIndex(TodoTable.Cols.PARENTS));
    mChildren = cursor.getString(cursor.getColumnIndex(TodoTable.Cols.CHILDREN));
  }
```

I wouldn't copy that exactly, just giving you an idea. Also this is a good opportunity for some tests! You should add a `hashCode` and `equals()` method to your `TodoItem` POJO and you can create different instances using different constructors and test that they are equal when you expect them to be equal :+1: